src/inmanta/agent/resourcepool.py

```diff
@@ -337,11 +346,17 @@ async def cleanup_inactive_pool_members_task(self) -> None:
         We split up `cleanup_inactive_pool_members` and `cleanup_inactive_pool_members_task` in order to be able to call the
         cleanup method in the test without being blocked in a loop.
         """
-        while self.running:
-            sleep_interval = await self.cleanup_inactive_pool_members()
-            if self.running:
-                LOGGER.log(LOG_LEVEL_TRACE, "Manager will clean in %.2f seconds", sleep_interval)
-                await asyncio.sleep(sleep_interval)
+        try:
+            while self.running:
+                sleep_interval = await self.cleanup_inactive_pool_members()
+                if self.running:
+                    LOGGER.log(LOG_LEVEL_TRACE, "Manager will clean in %.2f seconds", sleep_interval)
+                    # Allow wait to be cancelled on shutdown
+                    self.shutdown_sleep = asyncio.create_task(asyncio.sleep(sleep_interval))
+                    await self.shutdown_sleep
+        except CancelledError:
+            # We are woken up by shutdown
+            pass
```

Comment:
I don't like this at all.

It is too messy to be able to
1. wait for completion in the join method
2. prevent cancel of the cleanup itself. 

I wonder if I should step to something more like our scheduler? 

or am I missing something?

Reply:
This is fine for me. I don't have the feeling that the alternatives will be much better.